### PR TITLE
Problem: Unrecognized governance proposal types should not panic

### PR DESCRIPTION
**Pull request by vincentysc**
Issue: #707 
Solution: Return error on unrecognised gov proposal type

### CODE REVIEW

File: usecase/parser/msg_grant_stake_grant_test.go [hunk 1]
```diff
@@ -60,9 +61,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_grant_stake_grant_test.go:68]: some minor indentation issue

File: usecase/parser/msg_ibc_channel_open_ack_test.go [hunk 2]
```diff
@@ -56,9 +57,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_channel_open_ack_test.go:64]: here as well

File: usecase/parser/msg_ibc_channel_open_confirm_test.go [hunk 3]
```diff
@@ -55,9 +56,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_channel_open_confirm_test.go:63]: here as well

File: usecase/parser/msg_ibc_channel_open_init_test.go [hunk 4]
```diff
@@ -60,9 +61,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_channel_open_init_test.go:68]: here as well

File: usecase/parser/msg_ibc_channel_open_try_test.go [hunk 5]
```diff
@@ -67,9 +68,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_channel_open_try_test.go:75]: here as well

File: usecase/parser/msg_ibc_connection_open_ack_test.go [hunk 6]
```diff
@@ -138,9 +139,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_connection_open_ack_test.go:146]: here as well

File: usecase/parser/msg_ibc_connection_open_init_test.go [hunk 7]
```diff
@@ -56,9 +57,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_connection_open_init_test.go:64]: here as well

File: usecase/parser/msg_ibc_connection_open_confirm_test.go [hunk 8]
```diff
@@ -54,9 +55,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_connection_open_confirm_test.go:62]: here as well

File: usecase/parser/msg_ibc_connection_open_try_test.go [hunk 9]
```diff
@@ -146,9 +147,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_connection_open_try_test.go:154]: here as well

File: usecase/parser/msg_ibc_transfer_transfer_test.go [hunk 10]
```diff
@@ -68,9 +69,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_transfer_transfer_test.go:76]: here as well

File: usecase/parser/msg_ibc_create_tendermint_client_test.go [hunk 11]
```diff
@@ -109,9 +110,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_create_tendermint_client_test.go:117]: here as well

File: usecase/parser/msg_ibc_create_solomachine_client_test.go [hunk 12]
```diff
@@ -64,9 +65,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_create_solomachine_client_test.go:72]: here as well

File: usecase/parser/msg_ibc_update_tendermint_client_test.go [hunk 13]
```diff
@@ -148,9 +149,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			stakingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
+			logger := test.NewFakeLogger()
 
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+        logger,
```
Comment (calvinaco) [inline usecase/parser/msg_ibc_update_tendermint_client_test.go:156]: here as well

File: usecase/parser/msg_nft_burn_nft_test.go [hunk 14]
```diff
@@ -25,9 +26,11 @@ var _ = Describe("ParseMsgCommands", func() {
 			bondingDenom := "basecro"
 
 			pm := usecase_parser_test.InitParserManager()
-
+			logger := test.NewFakeLogger()
+			
 			cmds, err := parser.ParseBlockTxsMsgToCommands(
 				pm,
+                logger,
```
Comment (calvinaco) [inline usecase/parser/msg_nft_burn_nft_test.go:33]: here as well

File: usecase/parser/utils/parsermanager.go [hunk 15]
```diff
@@ -46,6 +46,7 @@ type CosmosParserParams struct {
 	MsgIndex        int
 	Msg             map[string]interface{}
 	ParserManager   *CosmosParserManager
+	Logger   		applogger.Logger
```
Comment (calvinaco) [inline usecase/parser/utils/parsermanager.go:49]: tab and space mixed here

File: usecase/parser/msg.go [hunk 16]
```diff
@@ -322,9 +325,8 @@ func ParseMsgSubmitProposal(
 		cmds = parseMsgSubmitCancelSoftwareUpgradeProposal(parserParams.MsgCommonParams.TxSuccess, parserParams.TxsResult, parserParams.MsgIndex, parserParams.MsgCommonParams, parserParams.Msg, rawContent)
 	} else if proposalContent.Type == "/cosmos.gov.v1beta1.TextProposal" {
 		cmds = parseMsgSubmitTextProposal(parserParams.MsgCommonParams.TxSuccess, parserParams.TxsResult, parserParams.MsgIndex, parserParams.MsgCommonParams, parserParams.Msg, rawContent)
-		// FIXME: https://github.com/crypto-com/chain-indexing/issues/707
-		//} else {
-		//	panic(fmt.Sprintf("unrecognzied govenance proposal type `%s`", proposalContent.Type))
+		} else {
+		parserParams.Logger.Errorf("unrecognzied govenance proposal type `%s`", proposalContent.Type)
```
Comment (calvinaco) [inline usecase/parser/msg.go:329]: not indented?

**Review by calvinaco — CHANGES_REQUESTED** [review]
overall looks good to me. There are some indentation issue though. May need to check your IDE settings to align with the project

**Comment by calvinaco** [comment]
Extending from this, I am thinking we can have a raw proposal message type that stores the raw JSON of unknown proposals. This PR looks good to me, but maybe we can add this to the backlog @davcrypto what do you think.

**Review by calvinaco — APPROVED** [review]

File: usecase/parser/block.go [hunk 17]
```diff
@@ -12,6 +13,7 @@ import (
 
 func ParseBlockToCommands(
 	parserManager *utils.CosmosParserManager,
+	logger applogger.Logger,
```
Comment (davcrypto) [inline usecase/parser/block.go:16]: Do you think it make sense to use the `Logger` from parserManager rather a new one?

**Comment by davcrypto** [comment]
> Extending from this, I am thinking we can have a raw proposal message type that stores the raw JSON of unknown proposals. This PR looks good to me, but maybe we can add this to the backlog @davcrypto what do you think.

i think we can just store the proposal JSON in the data field on view_ proposal, but lets open another issue for this.

File: usecase/parser/block.go (hunk 17 above)
Comment (vincentysc) [reply usecase/parser/block.go:16]: Agree, I update it.

**Pull request merged**